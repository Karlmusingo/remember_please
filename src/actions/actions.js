import axios from "axios";

export const fetchEvents = () => async dispatch => {
  const { data: payload } = await axios.get(
    `${process.env.BACKEND_URL}/events/`
  );

  return dispatch({
    type: "FETCH_EVENTS",
    payload
  });
};
export const addEvent = event => async () => {
  try {
    const day = event.date.split("/")[0];
    const month = event.date.split("/")[1];
    const year = event.date.split("/")[2] || 2019;
    const target = event.name;
    const phonenumber = event.whatsappNumber;
    const data = {
      day,
      month,
      year,
      target,
      message: event.message,
      phonenumber,
      country: "Rwanda",
      notificationTime: 7,
      type: "Birthday"
    };

    const res = await axios.post(`${process.env.BACKEND_URL}/events/`, data);

    return res.data;
  } catch (error) {
    return { error: "Oops" };
  }
};
