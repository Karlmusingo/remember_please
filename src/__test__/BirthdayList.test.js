import React from "react";
import { shallow, mount } from "enzyme";
import mockAxios from "axios";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import promiseMiddleware from "redux-promise-middleware";
import { fetchEvents, addEvent } from "../actions/actions";
import BirthdayList from "../components/birthdayList";
import { jsxText } from "@babel/types";

const mockStore = configureMockStore([thunk]);

describe("Birthday List component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      events: {}
    });
  });

  test("Should match the snapshot", async () => {
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({
        data: { id: 1, name: "karl" }
      })
    );

    const events = [
      {
        date: "12/04/2019",
        name: "karl",
        message: "message",
        whatsappNumber: "+23453456456"
      }
    ];

    await store.dispatch(fetchEvents(events));
    const actions = store.getActions();

    expect.assertions(3);
    expect(actions[0].type).toEqual("FETCH_EVENTS");
    expect(actions[0].payload).toEqual("mocked");

    expect(1).toBe(1);
  });
  test("Should match the snapshot", async () => {
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({
        data: { id: 1, name: "karl" }
      })
    );

    const events = {
      date: "12/04/2019",
      name: "karl",
      message: "message",
      whatsappNumber: "+23453456456"
    };
    const result = await store.dispatch(addEvent(events));

    // expect.assertions(3);
    expect(result.id).toEqual(1);
    expect(result.name).toEqual("karl");

    expect(1).toBe(1);
  });

  test("Should match the snapshot", async () => {
    mockAxios.post.mockImplementationOnce(() =>
      Promise.reject({
        error: "Oops ):"
      })
    );

    const events = {
      name: "karl",
      message: "message",
      whatsappNumber: "+23453456456"
    };
    const result = await store.dispatch(addEvent(events));

    // expect.assertions(3);
    expect(result.error).toEqual("Oops");

    expect(1).toBe(1);
  });
});

describe("BirthdayList snapshot", function() {
  let store;

  beforeEach(() => {
    store = mockStore({
      events: {}
    });
  });
  test("Should match the snapshot", () => {
    const wrapper = shallow(<BirthdayList store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe("BirthdayList snapshot with props", function() {
  let store;

  const events = {
    todays: {
      events: [
        {
          id: 2,
          Messages: [
            {
              draft: true
            },
            {
              draft: false
            }
          ]
        }
      ],
      eventsCount: 1
    },
    thisWeek: {
      events: [
        {
          id: 1,
          Messages: [
            {
              draft: true
            },
            {
              draft: false
            }
          ]
        }
      ],
      eventsCount: 1
    },
    other: {
      events: [
        {
          id: 3,
          Messages: [
            {
              draft: true
            },
            {
              draft: false
            }
          ]
        }
      ],
      eventsCount: 0
    }
  };

  beforeEach(() => {
    store = mockStore({
      events,
      onFetchEvents: jest.fn()
    });
  });
  test("Should match the snapshot", () => {
    const wrapper = mount(<BirthdayList store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Should click the create button", () => {
    const historyMock = { push: jest.fn() };
    // const onClickSpy = jest.fn();
    const wrapper = mount(<BirthdayList store={store} history={historyMock} />);

    wrapper
      .find("#create")
      .at(0)
      .simulate("click");
    expect(historyMock.push).toHaveBeenCalled();
  });
});
