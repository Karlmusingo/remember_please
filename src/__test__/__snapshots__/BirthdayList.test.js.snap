// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirthdayList snapshot Should match the snapshot 1`] = `
<ContextProvider
  value={null}
>
  <BirthdayList
    events={Object {}}
    onFetchEvents={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`BirthdayList snapshot with props Should match the snapshot 1`] = `
<Connect(BirthdayList)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BirthdayList
    events={
      Object {
        "other": Object {
          "events": Array [
            Object {
              "Messages": Array [
                Object {
                  "draft": true,
                },
                Object {
                  "draft": false,
                },
              ],
              "id": 3,
            },
          ],
          "eventsCount": 0,
        },
        "thisWeek": Object {
          "events": Array [
            Object {
              "Messages": Array [
                Object {
                  "draft": true,
                },
                Object {
                  "draft": false,
                },
              ],
              "id": 1,
            },
          ],
          "eventsCount": 1,
        },
        "todays": Object {
          "events": Array [
            Object {
              "Messages": Array [
                Object {
                  "draft": true,
                },
                Object {
                  "draft": false,
                },
              ],
              "id": 2,
            },
          ],
          "eventsCount": 1,
        },
      }
    }
    onFetchEvents={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div>
      <Navbar
        title="Birthday List"
      >
        <div
          className="container"
          style={
            Object {
              "alignText": "center",
              "background": "#ffc3b8",
              "color": "#fff",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "display": "flex",
                "justifyContent": "space-between",
                "padding": "10px 25px 0px 20px",
              }
            }
          >
            <div
              style={
                Object {
                  "color": "#fff",
                }
              }
            >
              <p
                id="btn-cancel"
                onClick={[Function]}
              >
                Cancel
              </p>
            </div>
            <div>
              <span>
                 
                Birthday List
                 
              </span>
            </div>
          </div>
        </div>
      </Navbar>
      <div
        className="container"
      >
        <div
          style={
            Object {
              "padding": "5px",
            }
          }
        >
          <b>
            TODAY
          </b>
           
          <span
            className="badge badge-success text-sm"
          >
            1
          </span>
        </div>
        <Birthday
          key="2"
          status="done"
        >
          <div
            style={
              Object {
                "display": "flex",
                "justifyContent": "space-between",
              }
            }
          >
            <span>
              John Doe
            </span>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "green",
                }
              }
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-check-circle fa-w-16 "
                data-icon="check-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "green",
                  }
                }
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </div>
        </Birthday>
        <div
          style={
            Object {
              "padding": "5px",
            }
          }
        >
          <b>
            TOMORROW
          </b>
           
          <span
            className="badge badge-success text-sm"
          >
            1
          </span>
        </div>
        <Hr>
          <hr
            style={
              Object {
                "backgroundColor": "#fff",
                "color": "#f5f5f5",
                "margin": "1px",
              }
            }
          />
        </Hr>
        <Birthday
          key="1"
          status="done"
        >
          <div
            style={
              Object {
                "display": "flex",
                "justifyContent": "space-between",
              }
            }
          >
            <span>
              John Doe
            </span>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "green",
                }
              }
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-check-circle fa-w-16 "
                data-icon="check-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "green",
                  }
                }
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </div>
        </Birthday>
        <div
          style={
            Object {
              "padding": "5px",
            }
          }
        >
          <b>
            THIS MONTH
          </b>
           
          <span
            className="badge badge-success text-sm"
          >
            0
          </span>
        </div>
        <Hr>
          <hr
            style={
              Object {
                "backgroundColor": "#fff",
                "color": "#f5f5f5",
                "margin": "1px",
              }
            }
          />
        </Hr>
        <Birthday
          key="3"
          status="done"
        >
          <div
            style={
              Object {
                "display": "flex",
                "justifyContent": "space-between",
              }
            }
          >
            <span>
              John Doe
            </span>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "green",
                }
              }
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-check-circle fa-w-16 "
                data-icon="check-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "color": "green",
                  }
                }
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </div>
        </Birthday>
      </div>
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="plus-circle"
          id="create"
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          style={
            Object {
              "bottom": "15px",
              "color": "red",
              "fontSize": "50px",
              "position": "fixed",
              "right": "20px",
            }
          }
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-plus-circle fa-w-16 "
            data-icon="plus-circle"
            data-prefix="fas"
            focusable="false"
            id="create"
            onClick={[Function]}
            role="img"
            style={
              Object {
                "bottom": "15px",
                "color": "red",
                "fontSize": "50px",
                "position": "fixed",
                "right": "20px",
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </div>
    </div>
  </BirthdayList>
</Connect(BirthdayList)>
`;
